src/main/java/rx/schedulers/TrampolineScheduler.java

```diff
@@ -71,14 +71,17 @@ private Subscription enqueue(Action0 action, long execTime) {
                 return Subscriptions.unsubscribed();
             }
             final TimedAction timedAction = new TimedAction(action, execTime, COUNTER_UPDATER.incrementAndGet(TrampolineScheduler.this));
-            queue.add(timedAction);
+            synchronized (queue) {
+                queue.add(timedAction);
+            }
 
             if (wip.getAndIncrement() == 0) {
                 do {
-                    TimedAction polled = queue.poll();
-                    // check for null as it could have been unsubscribed and removed
-                    if (polled != null) {
-                        polled.action.call();
+                    synchronized (queue) {
+                        if (!queue.isEmpty()) {
+                            TimedAction polled = queue.poll();
+                            polled.action.call();
```

Comment:
Good call. I fixed the wrapping. Now I'm just using poll instead of isEmpty, since poll returns null iff isEmpty.